src/store/allCoins/index.js

```diff
@@ -96,39 +134,28 @@ function allCoinsReducer(state = initialState, action) {
         volumeError: false,
       }
     case SORT_BY:
-      let sortAsc
-      switch (action.payload) {
-        case 'id':
-          sortAsc = 'idSortAsc'
-          break
-        case 'current_price':
-          sortAsc = 'priceSortAsc'
-          break
-        case 'price_change_percentage_1h_in_currency':
-          sortAsc = 'hourSortAsc'
-          break
-        case 'price_change_percentage_7d_in_currency':
-          sortAsc = 'sevenDaySortAsc'
-          break
-        case 'price_change_percentage_24h_in_currency':
-          sortAsc = 'twentyFourHourSortAsc'
-          break
-        default:
-          sortAsc = 'name'
-      }
+      const configProperty = Object.entries(state.config).find(
+        (ele) => ele[1].sortBy === action.payload,
+      )[0]
 
       return {
         ...state,
         coinsData: state.coinsData.sort((a, b) =>
-          state[sortAsc]
+          state.config[configProperty]['sortByAsc']
             ? a[action.payload] < b[action.payload]

```

Comment:
maybe put this in a variable so you don't repeat yourself

const { payload } = action;
a[payload] < b[payload]

Reply:
Done